### DISCUSSION TITLE
Unable to access GITHUB_TOKEN

### DISCUSSION THREAD

**Discussion by paramt** [community/community#25525] [Actions]
<p>I’m trying to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#permissions-for-the-github_token" rel="nofollow noopener">use the GITHUB_TOKEN to authenticate on behalf of GitHub Actions</a>. I’m trying to access the “GITHUB_TOKEN” environment variable through a Python file, which is run when the action is triggered. But for some reason, it seems like the environment variable doesn’t exist.</p>
<p>My Python file is using the PyGithub package to authenticate:</p>
<pre><code>import os
from github import Github

github = Github(os.environ["GITHUB_TOKEN"])
</code></pre>
<p>and it throws this error:</p>
<pre><code>Traceback (most recent call last):
  File "add_url.py", line 4, in &lt;module&gt;
    github = Github(os.environ["GITHUB_TOKEN"])
  File "/usr/lib/python2.7/UserDict.py", line 40, in __getitem__
    raise KeyError(key)
KeyError: 'GITHUB_TOKEN'
##[error]Process completed with exit code 1
</code></pre>
<p>Am I misinterpreting the docs? <a href="https://github.com/paramt/go.param.me/commit/c1afabf56599c3d917f1a7394ef5303bfb9ce413/checks?check_suite_id=356403913" rel="nofollow noopener">Here’s the full error log</a></p>

**Comment by dawidd6** [comment, marked as answer]
<p>You need to set the env variable explicitly:</p>
<p>env:<br>
GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}</p>
